src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java

```
@@ -1,7 +1,76 @@
 package ru.yandex.practicum.filmorate.controller;
 
-import org.springframework.web.bind.annotation.RestController;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.*;
+import ru.yandex.practicum.filmorate.exception.NotFoundException;
+import ru.yandex.practicum.filmorate.exception.ValidationException;
+import ru.yandex.practicum.filmorate.model.Film;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
 
 @RestController
+@RequestMapping("/films")
 public class FilmController {
+    private final HashMap<Long, Film> films = new HashMap<>();
+    private Long id = 0L;
+    public static final Integer MAX_DESCRIPTION_LENGTH = 200;
+    public static final LocalDate MIN_RELEASE_DATE = LocalDate.of(1895, 12, 28);
+    private static final Logger LOG = LoggerFactory.getLogger(FilmController.class);
+
+    @GetMapping
+    public List<Film> getFilms() {
+        LOG.info("Пришел GET запрос /films");
+        List<Film> filmList = new ArrayList<>(films.values());
+        LOG.info("Отправлен ответ GET /films с телом: {}", filmList);
+        return filmList;
+    }
+
+    @PostMapping()
+    public Film addFilm(@RequestBody Film film) {
+        filmValidation(film);
+        film.setId(++id);
+        films.put(film.getId(), film);
+        return film;
+    }
+
+    @PutMapping
+    public Film updateFilm(@RequestBody Film film) {
+        LOG.info("Пришел PUT запрос /films с телом: {}", film);
+        Long filmId = film.getId();
+        if (films.containsKey(filmId)) {
```

Лучше сделать наоборот, if (!films.containsKey(filmId)) {throw new NotFoundException("Фильм с id " + film.getId() + " не найден");} А дальше уже весь основной код. Так основной код не будет сдвинут и не нужно будет искать, где заканчивается if. Это так называемый fail fast.